debezium-server-core/src/main/java/io/debezium/server/DebeziumMetrics.java

```diff
@@ -33,16 +33,21 @@ public class DebeziumMetrics {
     public static final MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
 
     private ObjectName snapshotMetricsObjectName;
+    private ObjectName snapshotPartitionMetricsObjectName;
     private ObjectName streamingMetricsObjectName;
+    private ObjectName streamingPartitionMetricsObjectName;
 
-    private static ObjectName getDebeziumMbean(String context) {
+    private static ObjectName getDebeziumMbean(String context, boolean partitioned) {
         ObjectName debeziumMbean = null;
 
         for (ObjectName mbean : mbeanServer.queryNames(null, null)) {
 
             if (mbean.getCanonicalName().contains("debezium.")
                     && mbean.getCanonicalName().contains("type=connector-metrics")
-                    && mbean.getCanonicalName().contains("context=" + context)) {
+                    && mbean.getCanonicalName().contains("context=" + context)
+                    && (mbean.getCanonicalName().contains("debezium.sql_server:")
```

Comment:
Could we extract this sub-condition to a separate method with suitable to better convey its meaning